nuxeo-services/login/nuxeo-platform-login-token-rest/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/login/tokenauth/AuthenticationTokensObject.java

```diff
@@ -65,12 +65,32 @@ protected void initialize(Object... args) {
     }
 
     @POST
+    @Produces(MediaType.TEXT_PLAIN)
     public Response createToken(@QueryParam("application") String applicationName,
             @QueryParam("deviceId") String deviceId, @QueryParam("deviceDescription") String deviceDescription,
             @QueryParam("permission") String permission) {
         String username = getCurrentUser().getName();
         String token = service.acquireToken(username, applicationName, deviceId, deviceDescription, permission);
-        return Response.ok(token).status(Response.Status.CREATED).build();
+        return Response.ok(token).type(MediaType.TEXT_PLAIN_TYPE).status(Response.Status.CREATED).build();
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createJSONToken(@QueryParam("application") String applicationName,
+            @QueryParam("deviceId") String deviceId, @QueryParam("deviceDescription") String deviceDescription,
+            @QueryParam("permission") String permission, @QueryParam("detail") boolean detail) {
+        String username = getCurrentUser().getName();
+        String token = service.acquireToken(username, applicationName, deviceId, deviceDescription, permission);
+        Object body = String.format("\"%s\"", token);
+        if (detail) {
+            // Get the token from the service to populate additional fields like 'creationDate'
+            AuthenticationToken at = getTokens(applicationName).stream().filter(item -> token.equals(item.getToken()))
+                    .findFirst().orElse(null);
+            if (at != null) {
+            	body = at;
+            }
```

Comment:
This could be reworked to:
```java
body = getTokens(applicationName).stream()
                                 .filter(item -> token.equals(item.getToken()))
                                 .findFirst()
                                 .orElse(body);
```

Reply:
It could, but the receiver type would have to be Object.  I kept it this way for type readability.

Reply:
It could be interesting if we're using the type, this is not the case here. The information it brings can be given by recent IDE on findFirst() line.

For readability too, the current has an extra null check making code strange to read. 
Also, it is not formatted, which will end to something less readable in my opinion:
```java
AuthenticationToken at = getTokens(applicationName).stream()
                                                   .filter(item -> token.equals(item.getToken()))
                                                   .findFirst()
                                                   .orElse(null);
if (at != null) {
    body = at;
}
```

Reply:
Resolved this issue, had to erase the type signature with a map call.